tests/coverage_lcov/test_lcov.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+set -eu
+# for info see 'read.me'
+
+##############################################################
+# Folders structure (20.08.2020)
+# NOTE: Drawing will be removed after approval script
+##############################################################
+# facelift/                 <-- directory with source code
+# ├─ build/
+# │  ├─                     <-- working directory
+# │  └─
+# ├─ ci/
+# │  ├─ appveyor
+# │  └─ travis
+# ├─ cmake/
+# ├─ CMakeLists.txt
+# ├─ src/
+# ├─ tests/
+# │  ├─ asyncfunctions/
+# │  ├─ benchmarking/
+# │  ├─ cmake/
+# │  ├─ CMakeLists.txt
+# │  ├─ combined/
+# │  ├─ coverage_lcov/
+# │  │   ├─ llvm-gcov.sh    <--- patch for LCOV
+# │  │   ├─ read.me
+# │  │   └─ test_lcov.sh*   <--- executable script
+
+function outerdir(){
+   echo "$1" | sed -E 's@/[^.]+.*@/@'
+}
+DIR_BUILD=$PWD
+DIR_OUTER=$(outerdir "$0")
+if [ "$DIR_OUTER" = "./" ]
+then
+     echo "The script must be run from the 'facelift/build'"
+     exit 1
+fi
+DIR_SRC=$(cd "$DIR_OUTER"; pwd -P)
+DIR_CI="$DIR_SRC/ci/travis"
+DIR_GTEST="$DIR_BUILD/gtest"
+DIR_GTEST_INSTALL="$DIR_GTEST/googletest/install"
+GTEST_SETUP="$DIR_CI/setup_gtest.sh"
+DATE_TIME=$(date +"%y-%m-%d-%H%M%S")
+DIR_REPORT="$DIR_BUILD/lcov/report_coverage_${DATE_TIME}"
+REPORT_COVERAGE="$DIR_REPORT/index.html"
+REPORT_CTEST="$DIR_REPORT/ctest.log"
+COVERAGE_FILE="${DIR_BUILD}/coverage.info"
+COVERAGE_OUT_FILE="${DIR_BUILD}/coverage_out.info"
+NUMBER_THREADS="-j 4"
+LLVM_GCOV="llvm-gcov.sh"
+
+# Install gtest
+if [ ! -d $DIR_GTEST ]
+  then
+    mkdir -p $DIR_GTEST
+    pushd $DIR_GTEST
+    # Setup test
+    . $GTEST_SETUP
+    popd
+fi
+CMAKE_SYSTEM_PREFIX_PATH="$DIR_GTEST_INSTALL"
+# Build facelift with flags
+cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_COMPILER=${COMPILER} -DCMAKE_EXE_LINKER_FLAGS="-lgcov --coverage" -DCMAKE_CXX_FLAGS="-fprofile-arcs -ftest-coverage" -DFACELIFT_BUILD_TESTS=ON -DCMAKE_SYSTEM_PREFIX_PATH=$CMAKE_SYSTEM_PREFIX_PATH $DIR_SRC && cmake --build ${DIR_BUILD} -- $NUMBER_THREADS
```

Comment:
what is the ${COMPILER}?
Do you create coverage report for each compiler? We don't want to do that. Pick one.